router.go

```diff
@@ -158,6 +163,10 @@ func (m MapRouter) Choose(ctx context.Context, req *transport.Request) (transpor
 		service = m.defaultService
 	}
 
+	if _, ok := m.allServices[service]; !ok {
```

Comment:
Keep this as is (it will still work even after my suggestion above).

But FYI, if we kept the data structure as a map from service name to a `bool`, we could just do:
```go
if m.allServices[service] {
 // ...
}
```
since we can rely on go defaults. eg: https://play.golang.org/p/UYTGXp_kMki